fill out xit tests & logic for empty response body

ianknauer opened the pull request:
Filled out the two xit tests that were within the REPL spec.js file. 

First was a simple case to show that it included the error message that already existed.
Second including building out some logic to show/hide an error message when the body provided isn't in the correct JSON format. 

I styled the error to match the error shown when the URL isn't filled out.

<img width="945" alt="Screen Shot 2020-05-28 at 1 49 48 PM" src="https://user-images.githubusercontent.com/2685721/83192299-2adadf80-a0ea-11ea-89db-f7f8e464e3ce.png">

ianknauer commented on package.json:
```diff
@@ -18,7 +18,7 @@
     "percy:exec": "percy exec",
     "cy:run": "cypress run",
-    "cy:open": "cypress open --config baseUrl=http://localhost:8000",
+    "cy:open": "cypress open --config baseUrl=http://localhost:9000",
```
This was necessary to get the test:cy:run command to run correctly

ianknauer commented on package.json:
```diff
@@ -62,6 +62,7 @@
     "gatsby-transformer-sharp": "^2.5.3",
     "hex-rgb": "^4.1.0",
     "install": "^0.13.0",
+    "json5": "^2.1.3",
```
Added json5 to support json verification for strings that didn't add quotation marks around some of the attributes.

The standard JSON.parse() was failing on any of the tests that were formatted as follows

` {
    user: {
       name: 'Samuel'
     }
}`

samselikoff commented:
Looks great! Thanks so much for this!

I think we should allow empty string, and not differentiate on DELETE. Because some APIs do take in a request body on DELETE (its part of the http spec) so we should treat it the same as post/patch/put.

I hit a "bug" when I tried a POST to /users/1 with empty body, got the warning, changed the selector to DELETE then was able to send the request (but the error didn't clear). But I think if you update it to allow for empty string + also not differentiate on DELETE, all should be good.

ianknauer commented:
> Looks great! Thanks so much for this!
> 
> I think we should allow empty string, and not differentiate on DELETE. Because some APIs do take in a request body on DELETE (its part of the http spec) so we should treat it the same as post/patch/put.
> 
> I hit a "bug" when I tried a POST to /users/1 with empty body, got the warning, changed the selector to DELETE then was able to send the request (but the error didn't clear). But I think if you update it to allow for empty string + also not differentiate on DELETE, all should be good.
> 
> Let me know if that's not clear!

@samselikoff All made sense 👍 I was curious why the field was showing for the DELETE, i have never interacted with an API that actually did anything with the request body on delete. 

I've switched the JSON validation function to allow for null values or empty strings, but continued checking for JSON validation if there's any content in the string.

samselikoff commented:
Works great! Thanks Ian! 👊
